### PR TITLE
Add README file for developers

### LINKED ISSUE #1: Add README.dev.md file
This file should describe how to setup the environment to develop the package. For instance, installing pre-commit.
Check https://github.com/abelsiqueira/PlaygroundWithGit.jl for example.

### PR DESCRIPTION
**Pull request by datejada**
# Pull request details

## Describe the changes made in this pull request

This file has the developers' guide for contributions to this repository.

## List of related issues or pull requests

Closes #1 

## Collaboration confirmation

As a contributor I confirm

- [x] I read and followed the instructions in README.dev.md
- [NA] The documentation is up to date with the changes introduced in this Pull Request (or NA)
- [x] Tests are passing
- [NA] Lint is passing

### CODE REVIEW

**Comment by abelsiqueira** [comment]
I think it would be better to add a lint workflow (#8) before this so we can check that it has the right format.

**Comment by datejada** [comment]
The pull request (#20) solves the issue (#8) of adding a lint workflow to the repository

File: README.dev.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Developer documentation
+
+- Fork this repository.
+- Clone your fork.
+- Add this repo as upstream with
```
Comment (nope82) [inline README.dev.md:5]:
```suggestion
1. Fork this repository.
2. Clone your fork.
3. Add this repo as upstream with the following command:
```

Comment (nope82) [reply README.dev.md:5]: I would put is as an enumeration list to give it more of a workflow feeling, what do you think?

File: README.dev.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+```bash
+git remote add upstream https://github.com/TNO-Tulipa/TulipaBulb.jl
+```
+
+- Open this project in your editor, activate and instantiate the project.
+- Run the tests to make sure that everything is working as expected.
```
Comment (nope82) [inline README.dev.md:12]:
```suggestion
5. Open this project in your editor. 
6. On Julia, activate and instantiate the project under the package mode.
7. Run the tests to make sure that everything is working as expected.
```

File: README.dev.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+```bash
+pre-commit run -a
+```
+
+Do it once now to make sure that everything works as expected.
+
+Now, you can only commit if all the pre-commit tests pass.
+
+**Note:** on subsequent occassions when you need to run pre-commit in a new shell, you will need to activate the Python virtual environment.
```
Comment (nope82) [inline README.dev.md:51]:
```suggestion
**Note:** on subsequent occassions when you need to run pre-commit in a new shell, you will need to activate the Python virtual environment. If so, do the following:
```

File: README.dev.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+## Contributing workflow
+
+- Always branch from `main`
+
+- Add a branch prefix
+
+  - If there is an associated issue, add the issue number
+  - If there is no associated issue, **and the changes are small**, add a prefix such as "typo", "hotfix", "small-refactor", according to the type of update.
+  - If the changes are not small and there is no associated issue, then create the issue first, so we can properly discuss the changes
+- Commit message
```
Comment (nope82) [inline README.dev.md:66]:
```suggestion

- Commit message
```

File: README.dev.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+  - Use imperative, present tense (Add feature, Fix bug)
+  - Have informative titles
+  - If necessary, add a body with details
+- Try to create "atomic git commits" (recommended reading: [The Utopic Git History](https://blog.esciencecenter.nl/the-utopic-git-history-d44b81c09593))
```
Comment (nope82) [inline README.dev.md:71]:
```suggestion

- Try to create "atomic git commits" (recommended reading: [The Utopic Git History](https://blog.esciencecenter.nl/the-utopic-git-history-d44b81c09593))
```

**Review by nope82 — CHANGES_REQUESTED** [review]
Hi Diego, I left just a very small comments to the great work you already did. I believe that

**Pull request merged**